File: Prep-2025/prog/0019.cpp
#include<bits/stdc++.h>
using namespace std;
int n;
int s[11],b[11];
int mn=INT_MAX;
int main(){
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cin>>n;
    for(int i=0;i<n;++i){
        cin>>s[i]>>b[i];
    }
    for(int i=1;i<(1<<n);++i){
        int ress=1;
        int resb=0;
        for(int j=0;j<n;++j){
            if(i&(1<<j)){
                ress*=s[j];
                resb+=b[j];
            }
        }
        mn=min(mn,abs(ress-resb));
    }
    cout<<mn;
    return 0;
}

File: python-prog/0019.py
import sys
n=int(sys.stdin.readline())
s=list()
b=list()
mn=1e9
for i in range(n):
    si,bi=map(int,sys.stdin.readline().split())
    s.append(si)
    b.append(bi)
for i in range(1,1<<n):
    ress=1
    resb=0
    for j in range(n):
        if i&(1<<j):
            ress*=s[j]
            resb+=b[j]
    mn=min(mn,abs(ress-resb))
print(mn)